Dockerfile

```diff
@@ -12,21 +12,42 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# About: This dockerfile builds the cip binary for use in auditor e2e tests.
+# About: This dockerfile builds the cip binary for auditor tests and production use.
+#
+# Usage: Since there are two variants to build, you must include the variant name during build-time.
+# cip production binary: 
+#   docker build --build-arg variant=cip /path/to/Dockerfile
+# test auditor:
+#   docker build --build-arg variant=auditor /path/to/Dockerfile
+
+# Determine final build variant [cip | auditor].
```

Comment:
Looking at this again I think the variant shouldn't be called `cip vs auditor` because technically the same binary gets created in either case (just with a different `ENTRYPOINT`). I think `prod vs test` is a better naming scheme and more to the point.

Reply:
Sure, that seems fair!